eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/ChannelTlv.scala

```diff
@@ -53,4 +53,46 @@ object AcceptChannelTlv {
   val acceptTlvCodec: Codec[TlvStream[AcceptChannelTlv]] = tlvStream(discriminated[AcceptChannelTlv].by(varint)
     .typecase(UInt64(0), variableSizeBytesLong(varintoverflow, bytes).as[UpfrontShutdownScript])
   )
-}
\ No newline at end of file
+}
+
+sealed trait FundingCreatedTlv extends Tlv
+
+object FundingCreatedTlv {
+  val fundingCreatedTlvCodec: Codec[TlvStream[FundingCreatedTlv]] = tlvStream(discriminated[FundingCreatedTlv].by(varint))
+}
+
+sealed trait FundingSignedTlv extends Tlv
+
+object FundingSignedTlv {
+  val fundingSignedTlvCodec: Codec[TlvStream[FundingSignedTlv]] = tlvStream(discriminated[FundingSignedTlv].by(varint))
+}
+
+sealed trait FundingLockedTlv extends Tlv
+
+object FundingLockedTlv {
+  val fundingLockedTlvCodec: Codec[TlvStream[FundingLockedTlv]] = tlvStream(discriminated[FundingLockedTlv].by(varint))
+}
+
+sealed trait ChannelReestablishTlv extends Tlv
+
+object ChannelReestablishTlv {
+  val channelReestablishTlvCodec: Codec[TlvStream[ChannelReestablishTlv]] = tlvStream(discriminated[ChannelReestablishTlv].by(varint))
```

Comment:
Hope we didn't mess up this one!

https://github.com/ACINQ/eclair/blob/44d37bfd9e1f054f226e42123a8a9a8b0e2badaa/eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageCodecs.scala#L62-L66

Reply:
This patch proves that our peer-backup codec is indeed tlv-compliant:

```diff
diff --git a/eclair-core/src/test/scala/fr/acinq/eclair/wire/protocol/LightningMessageCodecsSpec.scala b/eclair-core/src/test/scala/fr/acinq/eclair/wire/protocol/LightningMessageCodecsSpec.scala
index 1a59dd6a7..db02ad4ca 100644
--- a/eclair-core/src/test/scala/fr/acinq/eclair/wire/protocol/LightningMessageCodecsSpec.scala
+++ b/eclair-core/src/test/scala/fr/acinq/eclair/wire/protocol/LightningMessageCodecsSpec.scala
@@ -24,6 +24,7 @@ import fr.acinq.eclair.router.Announcements
 import fr.acinq.eclair.wire.protocol.LightningMessageCodecs._
 import fr.acinq.eclair.wire.protocol.ReplyChannelRangeTlv._
 import org.scalatest.funsuite.AnyFunSuite
+import scodec.DecodeResult
 import scodec.bits.{ByteVector, HexStringSyntax}

 import java.net.{Inet4Address, InetAddress}
@@ -93,6 +94,46 @@ class LightningMessageCodecsSpec extends AnyFunSuite {
     }
   }

+  test("nonreg generic tlv") {
+    val channelId = randomBytes32()
+    val signature = randomBytes64()
+    val key = randomKey()
+    val point = randomKey().publicKey
+    val randomData = randomBytes(42)
+    val tlvTag = UInt64(hex"47010000")
+
+    val refs = Map(
+      (hex"0023" ++ channelId ++ signature, hex"") -> FundingSigned(channelId, signature),
+      (hex"0023" ++ channelId ++ signature ++ hex"fe47010000 00", hex"") -> FundingSigned(channelId, signature, TlvStream[FundingSignedTlv](Nil, GenericTlv(tlvTag, ByteVector.empty) :: Nil)),
+      (hex"0023" ++ channelId ++ signature ++ hex"fe47010000 07 cccccccccccccc", hex"") -> FundingSigned(channelId, signature, TlvStream[FundingSignedTlv](Nil, GenericTlv(tlvTag, hex"cccccccccccccc") :: Nil)),+
+      (hex"0088" ++ channelId ++ hex"0001020304050607 0809aabbccddeeff" ++ key.value ++ point.value, hex"") -> ChannelReestablish(channelId, 0x01020304050607L, 0x0809aabbccddeeffL, key, point),
+      (hex"0088" ++ channelId ++ hex"0001020304050607 0809aabbccddeeff" ++ key.value ++ point.value ++ hex"fe47010000 00", hex"") -> ChannelReestablish(channelId, 0x01020304050607L, 0x0809aabbccddeeffL, key, point, TlvStream[ChannelReestablishTlv](Nil, GenericTlv(tlvTag, ByteVector.empty) :: Nil)),
+      (hex"0088" ++ channelId ++ hex"0001020304050607 0809aabbccddeeff" ++ key.value ++ point.value ++ hex"fe47010000 07 bbbbbbbbbbbbbb", hex"") -> ChannelReestablish(channelId, 0x01020304050607L, 0x0809aabbccddeeffL, key, point, TlvStream[ChannelReestablishTlv](Nil, GenericTlv(tlvTag, hex"bbbbbbbbbbbbbb") :: Nil)),
+
+      (hex"0084" ++ channelId ++ signature ++ hex"0000", hex"") -> CommitSig(channelId, signature, Nil),
+      (hex"0084" ++ channelId ++ signature ++ hex"0000 fe47010000 00", hex"") -> CommitSig(channelId, signature, Nil, TlvStream[CommitSigTlv](Nil, GenericTlv(tlvTag, ByteVector.empty) :: Nil)),
+      (hex"0084" ++ channelId ++ signature ++ hex"0000 fe47010000 07 cccccccccccccc", hex"") -> CommitSig(channelId, signature, Nil, TlvStream[CommitSigTlv](Nil, GenericTlv(tlvTag, hex"cccccccccccccc") :: Nil)),
+
+      (hex"0085" ++ channelId ++ key.value ++ point.value, hex"") -> RevokeAndAck(channelId, key, point),
+      (hex"0085" ++ channelId ++ key.value ++ point.value ++ hex" fe47010000 00", hex"") -> RevokeAndAck(channelId, key, point, TlvStream[RevokeAndAckTlv](Nil, GenericTlv(tlvTag, ByteVector.empty) :: Nil)),
+      (hex"0085" ++ channelId ++ key.value ++ point.value ++ hex" fe47010000 07 cccccccccccccc", hex"") -> RevokeAndAck(channelId, key, point, TlvStream[RevokeAndAckTlv](Nil, GenericTlv(tlvTag, hex"cccccccccccccc") :: Nil)),
+
+      (hex"0026" ++ channelId ++ hex"002a" ++ randomData, hex"") -> Shutdown(channelId, randomData),
+      (hex"0026" ++ channelId ++ hex"002a" ++ randomData ++ hex"fe47010000 00", hex"") -> Shutdown(channelId, randomData, TlvStream[ShutdownTlv](Nil, GenericTlv(tlvTag, ByteVector.empty) :: Nil)),
+      (hex"0026" ++ channelId ++ hex"002a" ++ randomData ++ hex"fe47010000 07 cccccccccccccc", hex"") -> Shutdown(channelId, randomData, TlvStream[ShutdownTlv](Nil, GenericTlv(tlvTag, hex"cccccccccccccc") :: Nil)),
+
+      (hex"0027" ++ channelId ++ hex"00000000075bcd15" ++ signature, hex"") -> ClosingSigned(channelId, 123456789.sat, signature),
+      (hex"0027" ++ channelId ++ hex"00000000075bcd15" ++ signature ++ hex"fe47010000 00", hex"") -> ClosingSigned(channelId, 123456789.sat, signature, TlvStream[ClosingSignedTlv](Nil, GenericTlv(tlvTag, ByteVector.empty) :: Nil)),
+      (hex"0027" ++ channelId ++ hex"00000000075bcd15" ++ signature ++ hex"fe47010000 07 cccccccccccccc", hex"") -> ClosingSigned(channelId, 123456789.sat, signature, TlvStream[ClosingSignedTlv](Nil, GenericTlv(tlvTag, hex"cccccccccccccc") :: Nil)),
+    )
+
+    refs.foreach { case ((bin, remainder), init) =>
+      assert(lightningMessageCodec.decode(bin.bits ++ remainder.bits).require === DecodeResult(init, remainder.bits))
+      assert(lightningMessageCodec.encode(init).require === bin.bits)
+    }
+  }
+
   test("encode/decode live node_announcements") {
     val ann = hex"a58338c9660d135fd7d087eb62afd24a33562c54507a9334e79f0dc4f17d407e6d7c61f0e2f3d0d38599502f61704cf1ae93608df027014ade7ff592f27ce2690001025acdf50702d2eabbbacc7c25bbd73b39e65d28237705f7bde76f557e94fb41cb18a9ec00841122116c6e302e646563656e7465722e776f726c64000000000000000000000000000000130200000000000000000000ffffae8a0b082607"
     val bin = ann.bits

```

Reply:
Added this test in https://github.com/ACINQ/eclair/pull/1891/commits/003c0db9cc9c6c1263fa5639bdb58c61e7e7641e